### PR TITLE
Add unit tests for Native Auth

### PR DESCRIPTION
**Pull request by SaurabhMSFT**
This PR adds unit tests for Native Auth. No new dependency has been added.

### CODE REVIEW

File: msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationJavaTest.java [hunk 1]
```diff
@@ -0,0 +1,2601 @@
...
+        state.signIn(null, callback);
+        assertTrue(resultFuture.get(30, TimeUnit.SECONDS) instanceof SignInResult.UnexpectedError);
+    }
+
+    /**
+     * Test sign in with SLT scenario 3:
+     * 1a -> sign in with (expired) SLT
+     * 1b <- server returns error
+     */
+    @Ignore("Waiting for SLT Mock API Integration (Out of scope?)")
```
Comment (fadidurah) [inline msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationJavaTest.java]: Has SLT support been added in the mock api yet?

File: msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt [hunk 2]
```diff
@@ -0,0 +1,2011 @@
...
+        val result = sltState.signIn(scopes = null)
+        assertTrue(result is SignInResult.UnexpectedError)
+    }
+
+    /**
+     * Test sign in with SLT scenario 2:
+     * 1a -> sign in with (expired) SLT
+     * 1b <- server returns error
+     */
+    @Ignore("Waiting for SLT Mock API Integration (Out of scope?)")
```
Comment (fadidurah) [inline msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt]: Same here, do we have mock api support for slt yet

**Review by fadidurah — APPROVED** [review]

File: msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt (hunk 2 above)
Comment (SaurabhMSFT) [reply msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt]: The SLT support is in the mock APIs. I will enable this test.

File: msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationJavaTest.java (hunk 1 above)
Comment (SaurabhMSFT) [reply msal/src/test/java/com/microsoft/identity/client/nativeauth/NativeAuthPublicClientApplicationJavaTest.java]: The SLT support is in the mock APIs. I will enable this test.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 3]
```diff
@@ -842,6 +842,8 @@ public static INativeAuthPublicClientApplication createNativeAuthPublicClientApp
                     null,
                     null
             );
+        }  catch (MsalException e) {
```
Comment (SaurabhMSFT) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java:845]: MsalException is a type of BaseException. The below clause was converting all MsalExceptions into unknown MsalException.

File: msal/src/androidTest/java/com/microsoft/identity/client/NativeAuthPublicApplicationKotlinTest.kt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client
```
Comment (SammyO) [inline msal/src/androidTest/java/com/microsoft/identity/client/NativeAuthPublicApplicationKotlinTest.kt]: Should be in dedicated native auth folder

Comment (SammyO) [reply msal/src/androidTest/java/com/microsoft/identity/client/NativeAuthPublicApplicationKotlinTest.kt]: and we should update code owners with this path

File: azure-pipelines/pull-request-validation/pr-msal.yml [hunk 5]
```diff
@@ -62,7 +62,13 @@ jobs:
   - task: Gradle@2
     displayName: Run Unit tests
     inputs:
-      tasks: msal:testLocalDebugUnitTest -Plabtest -ProbolectricSdkVersion=${{variables.robolectricSdkVersion}}
+      tasks: msal:testLocalDebugUnitTest -Plabtest -ProbolectricSdkVersion=${{variables.robolectricSdkVersion}} --tests *com.microsoft.identity.client*
```
Comment (SammyO) [inline azure-pipelines/pull-request-validation/pr-msal.yml:65]: @shahzaibj and @fadidurah can you review this? As this is a change to your test configuration, CI, etc.

Comment (fadidurah) [reply azure-pipelines/pull-request-validation/pr-msal.yml:65]: I haven't seen --tests used before, but i'm guessing you're just spliting the test suite into native auth and non-native auth, which seems fine to me

Comment (SaurabhMSFT) [reply azure-pipelines/pull-request-validation/pr-msal.yml:65]: yes, this runs tests in the specified package.

**Review by SammyO — APPROVED** [review]

**Pull request merged**